File: node_server/models/todo.model.js
const db = require('../config/db');
const mongoose = require('mongoose');

const { Schema } = mongoose;
const todoSchema = new Schema({
  title: {
    type: String,
    required: true,
  },
  desc: {
    type: String,
  },
  date: {
    type: String,
  },
  hour: {
    type: String,
  },
  isCompleted: {
    type: Boolean,
    default: false, 
  },
}, { timestamps: true });

const TodoModel = db.model('todo',todoSchema);
module.exports = TodoModel;

File: todo_app/lib/feature/model/todo_model.dart
class TodoModel {
  String? id;
  String? title;
  String? desc;
  String? date;
  String? hour;
  bool isCompleted;

  TodoModel({
    this.id,
    this.title,
    this.desc,
    this.date,
    this.hour,
    this.isCompleted = false,
  });

  TodoModel copyWith({
    String? id,
    String? title,
    String? desc,
    String? date,
    String? hour,
    bool? isCompleted,
  }) {
    return TodoModel(
      id: id ?? this.id,
      title: title ?? this.title,
      desc: desc ?? this.desc,
      date: date ?? this.date,
      hour: hour ?? this.hour,
      isCompleted: isCompleted ?? this.isCompleted,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'desc': desc,
      'date': date,
      'hour': hour,
      'isCompleted': isCompleted,
    };
  }

  factory TodoModel.fromJson(Map<String, dynamic> json) {
    return TodoModel(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      desc: json['desc'] ?? '',
      date: json['date'] ?? '',
      hour: json['hour'] ?? '',
      isCompleted: json['isCompleted'] ?? false,
    );
  }
}
